lib/project-helpers.ts

```diff
@@ -1,6 +1,50 @@
-export interface CreateProjectValues {
-  name: string;
-  goal: string;
-  owner: string;
-  champion: string;
+import { useEffect } from "react";
+import { useState } from "react";
+import { Project } from "../db/models/project";
+
+export class ProjectValues {
+  public id?: number;
+  public name: string = "";
+  public goal: string = "";
+  public owner: string = "";
+  public champion: string = "";
+  public adminEmails: string[] = [];
+
+  public constructor(init?: Partial<ProjectValues>) {
+    Object.assign(this, init);
+  }
+}
+
+export function useProject(id: string) {
+  const [project, setProject] = useState<ProjectValues>();
+  const [error, setError] = useState<string>();
+  const [isLoading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const getProject = async () => {
+      setLoading(true);
+
+      if (!id) return;
+
+      try {
+        const result = await fetch(`/api/v1/projects/${id}`);
+        const json = await result.json();
+        if (result.status === 200) {
+          const projectResult: ProjectValues = json;
```

Comment:
This and the json() function call can be done in a single assignment

Reply:
As per https://github.com/leggetter/habitdabit/pull/7#discussion_r1100035633, we only know the JSON structure and thus type once we've queried the `result.status`. So, we can't do the assignment to a type until we know the HTTP status code.

Reply:
Makes sense